### PR TITLE
Freeze account feat example cli

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
This PR adds a CLI for creating and sending transactions to a local testnet, that was used initially in testing the freeze account feature, but can be extended and used in other scenarios as well.

While writing the CLI, some missing changes on the transaction interactor were found and added.
These changes are required by any app that needs guarded txs handling.

### CODE REVIEW

File: interactors/interface.go [hunk 1]
```diff
@@ -22,3 +22,10 @@ type TxBuilder interface {
 	ApplyUserSignatureAndGenerateTx(skBytes []byte, arg data.ArgCreateTransaction) (*data.Transaction, error)
 	IsInterfaceNil() bool
 }
+
+// GuardedTxBuilder defines the component able to build and sign a guarded transaction
+type GuardedTxBuilder interface {
+	ApplyUserSignatureAndGenerateTx(skBytes []byte, arg data.ArgCreateTransaction) (*data.Transaction, error)
+	ApplyGuardianSignature(skGuardianBytes []byte, tx *data.Transaction) error
+	IsInterfaceNil() bool
+}
```
Comment (dragos-rebegea) [inline interactors/interface.go:31]: new line at the end of the file

File: cmd/cli/main.go [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+		var err error
+		if len(option) > 0 {
+			selectedAddress, err = data.NewAddressFromBech32String(option)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+	}
+
+	return selectedAddress, sk, nil
```
Comment (iulianpascalau) [inline cmd/cli/main.go:335]: for a provided, custom, address, you return the sk as nil here. (default case branch) is this ok?

File: cmd/cli/main.go [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+	withFunding    bool
+	send           bool
+	guardianSigned bool
+	guardedTxBy    string
+	guardian       string
+	sender         string
+	receiver       string
+	dataField      string
+	value          string
+	gasLimit       string
```
Comment (iulianpascalau) [inline cmd/cli/main.go:131]: Please switch this to uint64

File: cmd/cli/main.go [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+		log.Error("unable to load eve address from the private key", "error", err)
+		return nil, err
+	}
+
+	return td, nil
+}
+
+func createElrondProxyArgs() blockchain.ArgsElrondProxy {
+	return blockchain.ArgsElrondProxy{
+		ProxyURL:            examples.LocalTestnetGateway,
```
Comment (iulianpascalau) [inline cmd/cli/main.go:585]: please add a flag to define this

File: examples/keys.go [hunk 5]
```diff
@@ -6,3 +6,21 @@ NDEzZjQyNTc1ZjdmMjZmYWQzMzE3YTc3ODc3MTIxMmZkYjgwMjQ1ODUwOTgxZTQ4
 YjU4YTRmMjVlMzQ0ZThmOTAxMzk0NzJlZmY2ODg2NzcxYTk4MmYzMDgzZGE1ZDQy
 MWYyNGMyOTE4MWU2Mzg4ODIyOGRjODFjYTYwZDY5ZTE=
 -----END PRIVATE KEY for erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th-----`
+
+const BobPemContents = `-----BEGIN PRIVATE KEY for erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx-----
```
Comment (iulianpascalau) [inline examples/keys.go:10]: missing comments on exported items

File: examples/networks.go [hunk 6]
```diff
@@ -5,3 +5,6 @@ const MainnetGateway = "https://gateway.elrond.com"
 
 // TestnetGateway represents the testnet gateway address
 const TestnetGateway = "https://testnet-gateway.elrond.com"
+
+// LocalTestnetGateway represents a local testnet proxy address
+const LocalTestnetGateway = "http://localhost:7950"
```
Comment (iulianpascalau) [inline examples/networks.go:10]: this is too custom. Please remove this and instead use the TestnetGateway as the default one

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: interactors/interface.go (hunk 1 above)
Comment (AdoAdoAdo) [reply interactors/interface.go]: done

File: cmd/cli/main.go [hunk 7]
```diff
@@ -0,0 +1,593 @@
+package main
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply cmd/cli/main.go:1]: I think we can leave it as CLI, and simply add more functionality as needed.

File: cmd/cli/main.go [hunk 8]
```diff
@@ -0,0 +1,593 @@
...
+	addressEve     core.AddressHandler
+}
+
+func main() {
+	_ = logger.SetLogLevel("*:DEBUG")
+
+	app := cli.NewApp()
+	cli.AppHelpTemplate = helpTemplate
+
+	app.Name = "guarded transactions cli"
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply cmd/cli/main.go:160]: updated to make it more generic.

File: cmd/cli/main.go (hunk 2 above)
Comment (AdoAdoAdo) [reply cmd/cli/main.go:335]: In this case yes, if you use custom addresses, we do not have the private key for that so we return nil.
For the supported addresses (alice, bob, charlie, eve, etc) we have the private keys as well.

File: cmd/cli/main.go (hunk 3 above)
Comment (AdoAdoAdo) [reply cmd/cli/main.go:131]: done

File: cmd/cli/main.go (hunk 4 above)
Comment (AdoAdoAdo) [reply cmd/cli/main.go:585]: done

File: examples/networks.go (hunk 6 above)
Comment (AdoAdoAdo) [reply examples/networks.go:10]: removed, there is the option to give custom url for the proxy flag which should cover this as well.

File: cmd/cli/main.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply cmd/cli/main.go:1]: ok

**Review by iulianpascalau — APPROVED** [review]

File: cmd/cli/main.go [hunk 9]
```diff
@@ -0,0 +1,613 @@
...
+   {{range .Authors}}{{ . }}{{end}}
+   {{end}}{{if .Commands}}
+GLOBAL OPTIONS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}
+VERSION:
+   {{.Version}}
+   {{end}}
+`
+	setGuardian = cli.BoolFlag{
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply cmd/cli/main.go]: merged

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**